File: _REPO/APRESS/distributed-.net-programming-in-vb-.net/Chapter05/RemoteCallbackDelegate/ClientLibrary/SimpleMathResult.vb
Imports System.Threading

Public Class SimpleMathResult
   Inherits MarshalByRefObject
   
   Public Sub MathCallback(result As Integer)
      
      'Display the current thread ID
      Console.WriteLine("MathCallback executing on thread {0}", _
         Thread.CurrentThread.GetHashCode())

      Console.WriteLine("Add result is {0}", result)
   End Sub
End Class

File: _REPO/APRESS/distributed-.net-programming-in-csharp/Chapter05/RemoteCallbackDelegate/ClientLibrary/SimpleMathResult.cs
using System;
using System.Threading;

namespace ClientLibrary
{
   public class SimpleMathResult : MarshalByRefObject
   {
      public void MathCallback(int result)
      {
         // Display the current thread ID
         Console.WriteLine("MathCallback executing on thread {0}", 
            Thread.CurrentThread.GetHashCode());

         Console.WriteLine("Add result is {0}", result);

         // Simulate a long running callback
         Thread.Sleep(5000);
      }
   }
}
